### PR TITLE
New layout for tests folder and Isolation of broken tests

### PR DESCRIPTION
**Pull request by amenadiel**
- adds test for issue #593 
 - npm start now runs test/_server.js (separate process to avoid having each test spawning its own server)
 - adds .editorconfig to uniform 2 spaces and lf
 - fixes object on _svgs.js moving broken and passing tests under their respective groups
 - remove examples/index.html
 - isolate failing issues
 - update README.md with jsdeliver hits and npm installs badges
 - update start and test tasks in package.json
 - fixed svgs with DOM errors (even browser couldn't read them)
 - broken tests are now skipped on failure
 - fixed expected result for outdated pngs
 - changes layout of test/index.html

**Important:**

Since the renderers are different, the result for the browser will often differ from the expected node result, therefore the tests will fail. To avoid this, I rather have browser tests skipping on failure and providing a useful message. This means the browser tests are referential and the mandatory ones are the node tests.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,13 +19,13 @@
   "browser": "dist/browser/canvg.min.js",
   "version": "1.5.2",
   "scripts": {
-    "start": "node test/browser.test.js",
+    "start": "node test/_server.js",
```
Comment (gabelerner) [inline package.json]: yea i was going to make a comment on this on the other PR, reads much better!

File: test/test.css [hunk 2]
```diff
@@ -0,0 +1,47 @@
+ body {
```
Comment (gabelerner) [inline test/test.css]: does the test need css to work? i guess maybe only `font-family` is important

Comment (amenadiel) [reply test/test.css]: heh no, not really, I was just looking at it from the browser and needed some styling, I'll remove that file

Comment (gabelerner) [reply test/test.css]: ya i removed most of the dom in the last PR since it's for test only :ambulance:

**Comment by amenadiel** [comment]
> first, i think the examples array should just move to the solved_issues or broken for test purposes. for the actual examples page, it's an nice demo of capabilities but for the test, it's either passed or failed.

Yeah that sounds sensible. Let's keep the examples just like they are right now and split the svgs in passing/broken. 


> broken and solved seem really close in code, with the only difference being skip or fail right? can we go back to one test file for node and one for browser that, based on the broken or solved top level key, skips or fails the test?

I'll look into it. The browser test is really always the same. As I said, browser can pull out unexpected shenanigans on each chromium release so I'd stick to keeping node tests passing (unless they are already broken). So I could cut it down to 3 files: node.passing, node.broken and browser.everything. Is that OK?

**Comment by gabelerner** [comment]
why would node passing be a diff file from node broken? shouldn't it be just iterate over all svgs in the passing key and run the code and call `fail` and then iterate over all the svgs in the broken key and run the code and call `skip`.  the only difference between the 2 files would be that one call?

rest sounds good

**Comment by amenadiel** [comment]
> why would node passing be a diff file from node broken? shouldn't it be just iterate over all svgs in the passing key and run the code and call fail and then iterate over all the svgs in the broken key and run the code and call skip. the only difference between the 2 files would be that one call?

I can put everything in a single file and run two loops:

```js
for(let file in svgs['passing]) {
   t.truthy(...);
}
for(let file in svgs['broken']) {
   t.truthy.skip(...);
}
```

OR I can put `if` blocks inside one loop

```js
for( let group in svgs) {
   for (let file in svgs[group]) {
     
    if(group === 'broken') {
       t.truthy.skip(...);
    } else {
      t.truthy(...);
    }

   }
}
```

the second one seems dirty because there will be many nested ifs.

**Comment by amenadiel** [comment]
@gabelerner there you go, we're back to using just one test for node and other for browser.

File: test/index.html [hunk 3]
```diff
@@ -3,62 +3,115 @@
 
 <head>
   <title>canvg.js test</title>
+  <style>
```
[the remark this answers is not in the document]
Comment (amenadiel) [reply test/index.html]: Ok, I'll remove it and add file out of version control to perform visual comparison of failing issues.

File: test/node.test.js [hunk 4]
```diff
@@ -1,83 +1,84 @@
...
+});
 
-    test.serial(`comparing results for ${file} (${description})`, async t => {
 
-        let { canvasBuffer, expectedImg } = await getBuffersNode(file);
+for (let file in svgs['broken']) {
+  let description = svgs['broken'][file];
 
-        let actual_path = path.resolve(`${actual_folder}/${file}.png`);
+  test(`comparing results for ${file} (${description})`, async t => {
```
[the remark this answers is not in the document]
Comment (amenadiel) [reply test/node.test.js]: Yeah, but there are two places when that would be handled. Either when the file is too different from expected, or when the comparison throws. 

I'll do as you suggest in a few hours.

**Comment by amenadiel** [comment]
@gabelerner I restored the html files to their former state, and wrapped tests in a function to avoid doing two loops.

However, inside the function, I used the promise notation to have just one catch block capturing exceptions either from `canvg` or `blink-diff`. Having two try/catch blocks looked awful.

**Comment by gabelerner** [comment]
@amenadiel thanks for all this! merging!

i'll probably start working on issues soon and then maybe look at failing tests

**Pull request merged**